modules/javafx.controls/src/main/java/javafx/scene/control/SkinBase.java

```diff
@@ -238,7 +241,79 @@ protected final void registerChangeListener(ObservableValue<?> property, Consume
         return lambdaChangeListenerHandler.unregisterChangeListeners(property);
     }
 
+    /**
+     * Subclasses can invoke this method to register that they want to listen to
+     * invalidation events for the given observable. Registered {@link Consumer} instances
+     * will be executed in the order in which they are registered.
+     *
+     * @param observable the observable to observe for invalidation events
+     * @param consumer the consumer
+     * @since 17
+     */
+    protected final void registerInvalidationListener(Observable observable, Consumer<Observable> consumer) {
+        if (lambdaChangeListenerHandler == null) {
+            lambdaChangeListenerHandler = new LambdaMultiplePropertyChangeListenerHandler();
+        }
+        lambdaChangeListenerHandler.registerInvalidationListener(observable, consumer);
+    }
+
+    /**
+     * Unregisters all invalidation listeners that have been registered using
+     * {@link #registerInvalidationListener(Observable, Consumer)}
+     * for the given observable. The end result is that the given observable is no longer observed by any of the invalidation
+     * listeners, but it may still have additional listeners registered on it through means outside of
+     * {@link #registerInvalidationListener(Observable, Consumer)}.
+     *
+     * @param observable The observable for which all listeners should be removed.
+     * @return A single chained {@link Consumer} consisting of all {@link Consumer consumers} registered through
```

Comment:
1. There's no need for a `@link` on a class that is in the arguments or return of the method since they are linked there anyway, and it is recommended to use `@link` only the first time the class appears.

2. You might want to clarify that by "chained" you mean that they were composed using `Consumer#andThen`, either using `@plainlink` on "chained", or explicitly by "chained using Consumer#andThen`. Then again, it might be obvious.

Reply:
- kept only the link to the registerXX method (to clarify the scope of the removal)
- replaced chain by composed 
- concentrated on what that composed consumer does (performs all removed operations)

Not entirely certain whether it's clear enough now 

- should the description have an additional sentence `Returns a composed [same-as-in-returns]` Undecided.
- should the `same-as-in-returns` contain the specification of the sequence of performing (from the register method)? Tend to no (because it's getting too long), but then ..

Reply:
I haven't gone back and done a detailed review yet, but I like the overall changes. The one thing I did notice is that the language used to describe the return value of `unregisterInvalidationListeners` and `unregisterListChangeListeners` is different:

```
unregisterInvalidationListeners:
     * @return a composed consumer that performs all removed operations, or
     *  {@code null} if none have been registered or the observable is {@code null}
```

```
unregisterListChangeListeners:
     * @return A single chained {@link Consumer} consisting of all {@link Consumer consumers} registered through
     *      {@link #registerListChangeListener(ObservableList, Consumer)}. If no consumers have been registered on this
     *      list, null will be returned.
```

I find it confusing to say that the returned list "performs all removed operations", so I like the language in the latter method better. Some might interpret the former to mean that is is somehow necessary to call the returned chained consumer as part of the removal, which isn't what you meant to say.